### ISSUE TITLE
Assignment fails inside try-block

### ISSUE THREAD

**Issue by LuckyDino** [decaffeinate/decaffeinate#724]
decaffeinate is producing the wrong JavaScript based on my CoffeeScript input:

```coffee
xxx = try
  xxx = 1
```

[repl](http://decaffeinate-project.org/repl/#?evaluate=true&stage=EsnextStage&code=xxx%20%3D%20try%0A%20%20xxx%20%3D%201%0A)

I get this output:

```js
let xxx = (() => { try {
  return xxx = 1;
} catch (error) {} })();
```

Here's what coffee compiler produces:

```js
  var xxx;
  xxx = (function() {
    try {
      return xxx = 1;
    } catch (error) {}
  })();
```

Not sure if/how this should be fixed, but this caused bigger code block to fail on a codebase I was converting. The assignment inside the try-block was a leftover from some refactoring and should not have been there.

Decaffeinated code produces xxx = undefined, and with coffee compiler xxx = 1. This is because var xxx is defined before assignment in coffeescript, but decaffeinated code silently fails because xxx is not defined when the code is run and an exception is thrown inside the try-catch block.

**Comment by LuckyDino** [comment]
Ah, this seems to be duplicate of #585

**Comment by eventualbuddha** [comment]
Fixed in v2.41.2.
